src/Client.ts

```diff
@@ -174,9 +161,10 @@ export default class Client {
     opts?: Partial<ClientOptions>
   ): Promise<Client> {
     const options = defaultOptions(opts)
-    const waku = await createWaku(options)
-    const keys = await loadOrCreateKeysFromOptions(options, wallet, waku)
-    const client = new Client(waku, keys)
+    const apiClient = createApiClientFromOptions(options)
+    const keys = await loadOrCreateKeysFromOptions(options, wallet, apiClient)
+    apiClient.setAuthenticator(new Authenticator(keys.identityKey))
```

Comment:
I need the `apiClient` to get the keys from the network topic store, only once I have the keys can I add the authenticator. So, I have to go through this awkward flow of adding the authenticator later.